bundle/src/main/java/com/adobe/cq/wcm/contrib/components/internal/models/v1/HTMLContainerImpl.java

```diff
@@ -0,0 +1,238 @@
+
+                if (fileName != null) {
+                    // includeType will be null for HTML files...
+                    if (includeType == null || includeType.contentEquals(INLINE_TYPE)) {
+                        if (type.contentEquals(CSS_FILES)) {
+                            typedSB.append("<style type=\"text/css\">" + NL);
+
+                        } else if (type.contentEquals(JS_FILES)) {
+
+                            typedSB.append("<script type=\"text/javascript\">" + NL);
+
+                        }
+
+                        readFileIntoSB(fileName, typedSB);
+
+                        if (type.contentEquals(CSS_FILES)) {
+                            typedSB.append("</style>" + NL);
+
+                        } else if (type.contentEquals(JS_FILES)) {
+
+                            typedSB.append("</script>" + NL);
+
+                        }
+
+                    } else if (includeType.contentEquals(REFERENCE_TYPE) || includeType.contentEquals(DEFER_TYPE)
+                            || includeType.contentEquals(ASYNC_TYPE)) {
+                        if (type.contentEquals(CSS_FILES)) {
+                            typedSB.append("<link rel=\"stylesheet\" href=\"" + fileName + "\">");
+
+                        } else if (type.contentEquals(JS_FILES)) {
+
+                            typedSB.append("<script type=\"text/javascript\" "
+                                    + (String) (includeType.contentEquals(DEFER_TYPE)
+                                    || includeType.contentEquals(ASYNC_TYPE) ? includeType : "")
+                                    + " src=\"" + fileName + "\"></script>");
+                        }
+                    }
+                }
+            }
+        }
+        return (typedSB);
+    }
+
+    private StringBuffer readFileIntoSB(String fileName, StringBuffer sb) {
+
+        ResourceResolver resolver = resource.getResourceResolver();
+        if (resolver != null) {
+            Resource newResource = resolver.getResource(fileName);
+            if (newResource != null) {
+                Asset asset = newResource.adaptTo(Asset.class);
+                if (asset != null) {
+                    Resource original = asset.getOriginal();
+                    InputStream content = original.adaptTo(InputStream.class);
+                    if(content != null) {
+                        BufferedReader br = new BufferedReader(new InputStreamReader(content, StandardCharsets.UTF_8));
+                        try {
+                            String line;
+                            while ((line = br.readLine()) != null) {
+                                sb.append(line);
+                            }
+                            br.close();
+                        } catch (IOException e) {
+                            LOGGER.error("ERROR reading {} into the StringBuffer: {}", fileName, e.toString());
+                            e.printStackTrace();
```

Comment:
I would rather we just pass the exception to the logger and not print the stacktrace